classes/module/Module.php

```diff
@@ -1269,19 +1269,11 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
                         $file = substr($file, 0, -2);
                     }
 
-                    // We check any parse error before including the file.
-                    // If (false) is a trick to not load the class with "eval".
-                    // This way require_once will works correctly
-                    // But namespace and use statements need to be removed
-                    $content = preg_replace('/\n[\s\t]*?use\s.*?;/', '', $file);
-                    $content = preg_replace('/\n[\s\t]*?namespace\s.*?;/', '', $content);
                     try {
-                        if (substr(`php -l $file_path`, 0, 16) == 'No syntax errors' || eval('if (false){	'.$content.' }') !== false) {
-                            require_once(_PS_MODULE_DIR_.$module.'/'.$module.'.php');
-                        } else {
-                            throw new ParseError("Parse error");
-                        }
-                    } catch (ParseError $e) {
+                        $fileContents = _PS_MODULE_DIR_.$module.'/'.$module.'.php';
+                        $parser = (new PhpParser\ParserFactory)->create(PhpParser\ParserFactory::PREFER_PHP5);
```

Comment:
I think it depends on what most of our users might be using. Since most of them are reluctant to upgrade to PHP 7 I thought it would be better to use PHP5 in the beginning. What do you think?